### PR TITLE
Type aiocoap api

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- See https://github.com/home-assistant-libs/pytradfri/pull/388 for the previous attempt.
- I've corrected the type for the `Command.result` as that depends on the return value of the `process_result` callback.
- I'm planning to continue working on the command typing to make it more useful in follow up PRs.

### CODE REVIEW

File: pytradfri/api/aiocoap_api.py [hunk 1]
```diff
@@ -177,33 +198,22 @@ async def _execute(self, api_command, timeout):
 
         return api_command.result
 
-    def debug_comm(self, call_type, api_commands):
```
Comment (ggravlingen) [inline pytradfri/api/aiocoap_api.py:180]: This was only used to figure out the various types of returns we could get from the gateway I presume?

Comment (ggravlingen) [reply pytradfri/api/aiocoap_api.py:180]: Oh, I see now that you refactored it..

File: pytradfri/api/aiocoap_api.py [hunk 2]
```diff
@@ -15,56 +17,73 @@
...
-_SENTINEL = object()
+
+
+class UndefinedType(Enum):
+    """Singleton type for use with not set sentinel values."""
+
+    _singleton = 0
+
+
+_SENTINEL = UndefinedType._singleton  # pylint: disable=protected-access
```
Comment (ggravlingen) [inline pytradfri/api/aiocoap_api.py:35]: I have never seen this type of code before. What does it do?

Comment (MartinHjelmare) [reply pytradfri/api/aiocoap_api.py:35]: It's to type the singleton that is used to mark empty value.

https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions

Comment (ggravlingen) [reply pytradfri/api/aiocoap_api.py:35]: Thanks!

**Review by ggravlingen — APPROVED** [review]
Sweet!

**Pull request merged**